core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -117,11 +117,28 @@ class KafkaApis(val requestChannel: RequestChannel,
         case ApiKeys.DELETE_RECORDS => handleDeleteRecordsRequest(request)
         case ApiKeys.INIT_PRODUCER_ID => handleInitProducerIdRequest(request)
         case ApiKeys.OFFSET_FOR_LEADER_EPOCH => handleOffsetForLeaderEpochRequest(request)
-        case ApiKeys.ADD_PARTITIONS_TO_TXN => handleAddPartitionToTxnRequest(request)
-        case ApiKeys.ADD_OFFSETS_TO_TXN => handleAddOffsetsToTxnRequest(request)
-        case ApiKeys.END_TXN => handleEndTxnRequest(request)
-        case ApiKeys.WRITE_TXN_MARKERS => handleWriteTxnMarkersRequest(request)
-        case ApiKeys.TXN_OFFSET_COMMIT => handleTxnOffsetCommitRequest(request)
+        case ApiKeys.ADD_PARTITIONS_TO_TXN => handleWithInterBrokerVersionAtLeast(KAFKA_0_11_0_IV0, request, (error:Errors) => {
```

Comment:
Should all `ApiKey` really be tied to the inter broker protocol version? This is the first time in any request it has been checked. So that says to me that for most requests the inter broker protocol version is irrelevant. So is limiting the `ApiKeys` returned in `ApiVersionsResponse` based on inter broker protocol actually a valid thing to do?

Assuming that it is valid... What are we adding to ApiKeys to indicate the inter broker protocol version? `ApiVersion.id` ? `ApiVersion.messageFormatVersion`